### ISSUE TITLE
Alternative Search Engine / Manticore

### ISSUE THREAD

**Issue by ciur** [ciur/papermerge#473]
Currently Papermerge DMS supports only one search engine backend  - elasticsearch.
It would be great to support multiple search engines.
Consider [manticoresearch](https://github.com/manticoresoftware/manticoresearch).
Consider [Apache Solr](https://solr.apache.org/).
Consider [Typesense](https://typesense.org/).
Consider [PostgreSQL FTS](https://www.postgresql.org/docs/current/textsearch.html)

Any other option ?

**Comment by telsch** [comment]
The removed legacy database search engine for low resource hardware like raspberry pi?

**Comment by ciur** [comment]
That's actually a good point. I really enjoyed PostgreSQL full text search :).
The disadvantage of that approach is that it is specifically bound one database type - PostgreSQL, but on the other hand
it is easier to use as there is no need for an extra service.
PostgreSQL FTS will be added back.

**Comment by ciur** [comment]
These week I experimented with [Django Haystack](https://django-haystack.readthedocs.io/en/master/). I was pleasantly surprised - although packages looks like out f date - it is very well written and works surprisingly well.
When used it adds support to four backends: elasticsearch, solr, xapian and whoosh.
Solr and elasticsearch are well known and resource intensive.
Whoosh it python only search engine. First impression - it works rather well for small set of data.
I was completely amazed by Xapian. I never heard of it (until couple of days ago) - but it works surprisingly well, is well documented and has great python bindings.
And best of all - are you ready? - it is written in C++ and has very small memory footprint.
Long story short - [Xapian search](https://getting-started-with-xapian.readthedocs.io/en/latest/) is now included by default. No configuration required.
